pkg/podhooks/podhooks.go

```diff
@@ -15,15 +15,15 @@ type Callbacks struct {
 	PodCallbacks func(podInformer cache.SharedIndexInformer)
 }
 
-// RegisterCallbacksAtInit registers callbacks (should be called at init())
+// RegisterCallbacksAtInit registers callbacks. Must be called before InstallHooks.
```

Comment:
As I was writing the other comment, I remember that I tend to use `AtInit()` as a way to mark that this is expected to be called at `Init()` and so there is no effort to do synchronization. Maybe it's worth adding it in the comment as well?

So maybe:
```
// Must be called before InstallHooks and callers need to be serialized externally.
```

Since we are calling it from a single goroutine (`main`), I think we are find but might be worth documenting this.